File: Python/203.Remove Linked List Elements.py
#删除链表中的给定数字节点
class ListNode(object):
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution(object):
    def removeElements(self, head, val):
        nHead = ListNode(0)
        nHead.next = head
        head = nHead
        while  head.next:
            if head.next.val == val:
                head.next = head.next.next
            else:
                head = head.next
        return nHead.next

l1 = ListNode(6)
l2 = ListNode(2)
l3 = ListNode(6)
l4 = ListNode(3)
l5 = ListNode(5)
l6 = ListNode(4)
l7 = ListNode(6)
l1.next = l2
l2.next = l3
l3.next = l4
l4.next = l5
l5.next = l6
l6.next = l7

s = Solution()
p = s.removeElements(l1,6)
while p:
    print(p.val)
    p = p.next

File: C++/Linked List/203.Remove Linked List Elements.cpp
/*
删除链表中的某个值。
*/
#include <iostream>
using namespace std;

struct ListNode {
    int val;
    ListNode *next;
    ListNode(int x) : val(x), next(NULL) {}
};


class Solution {
public:
	ListNode* removeElements(ListNode* head, int val) {
        ListNode *nhead=new ListNode(0);
        nhead->next = head;
        ListNode *cur = nhead;
        while(cur->next)
        {
        	if(cur->next->val == val)
        	{
        		cur->next = cur->next->next;
        	}
        	else
        	{
        		cur = cur->next;
        	}
        }

        return nhead->next;
    }
};
int main()
{
	Solution s;
	ListNode head(3);
	ListNode t1(2);
	ListNode t2(8);
	ListNode t3(4);
	ListNode t4(3);
	ListNode t5(6);
	head.next = &t1;
	t1.next = &t2;
	t2.next = &t3;
	t3.next = &t4;
	t4.next = &t5;
 	ListNode* h = s.removeElements(&head,3);	
 	while(h)
 	{
 		cout<<h->val<<endl;
 		h = h->next;
 	}
}

